packages/components/src/internal/components/forms/QuerySelect.tsx

```diff
@@ -183,9 +183,14 @@ export class QuerySelect extends PureComponent<QuerySelectOwnProps, State> {
 
         try {
             const model = await initSelect(this.props);
+
+            if (!this.mounted) return;
+
             this.setState({ model });
         } catch (error) {
-            this.setState({ error });
+            if (!this.mounted) return;
+
+            this.setState({ error: error?.message ?? 'Failed to initialize.' });
```

Comment:
can/should this also use `resolveErrorMessage`?

Reply:
Done.